### PR TITLE
Heavy rework and cleanup of Factions & Users

### PR DESCRIPTION
**Pull request by ghost**
A heavy rewrite/cleaning/mass organization of various functions, tables, and general debauchery,

### CODE REVIEW

**Comment by sarahsturgeon** [comment]
Can one of the admins verify this patch?

**Comment by ghost** [comment]
That is a lot of conflicts. Oh boy

**Comment by ghost** [comment]
I think I got almost every conflict fixed now

**Comment by ghost** [comment]
Off the top of my head, some things fixed / adjusted in this big PR. 

1. Capitalized variable names in clientside classes. 
`self.internalname became self.InternalName`
2. Sorted tables by alphabetical order
3. Condense networking into a more cleaned (debatable ) state. 
4. Deleted most old or broken console commands
5. Cleaned and reworked sv_users
6. Cleaned and reworked sv_factions
7. Removed old/unused netvars
8. actually fixed permission system, working as intended now.

File: lua/cfcfactions/core/cl/dermas/cl_factionsderma.lua [hunk 1]
```diff
@@ -13,8 +13,7 @@ cfcFactions:RegisterDermaMenu( "View Factions", Panel, 1 )
 
 
 function Panel:Init()
-
-    cfcFactions.FactionsView = 1
+    cfcFactions.FactionsView=1
```
Comment (sarahsturgeon) [inline lua/cfcfactions/core/cl/dermas/cl_factionsderma.lua]:
```suggestion
    cfcFactions.FactionsView = 1
```

File: lua/cfcfactions/core/sv/sv_permsys.lua [hunk 2]
```diff
@@ -128,53 +128,82 @@ end
...
     end
 
-    if cfcuser:UserExists( player ) then
-        if ( not ( cfcuser[player:SteamID64()].CFCPermissions == nil ) ) then
+    if cfcuser:UserExists( authplayer ) then
+        if ( not ( cfcuser[authplayer:SteamID64()].CFCPermissions == nil ) ) then
             -- Error out, player already has proper permissions for authentication
             return
         end
     end
```
Comment (sarahsturgeon) [inline lua/cfcfactions/core/sv/sv_permsys.lua]: Not important, but would be nice to follow the camel-case convention, `authplayer` -> `authPlayer`

File: lua/cfcfactions/core/sv/sv_users.lua [hunk 3]
```diff
@@ -9,46 +9,49 @@ cfcFactions.Users = cfcFactions.Users or {}
...
 end
 
 -- Checks if a user is already registered
 function cfcuser:UserExists( user )
-
-    if not user:IsPlayer() then
-        -- Error out, not a player
-        return
-    end
+	if not IsValid( user ) and not user then
```
Comment (sarahsturgeon) [inline lua/cfcfactions/core/sv/sv_users.lua]:
```suggestion
	if not ( user and IsValid( user ) ) then 
```

File: lua/cfcfactions/core/sv/sv_users.lua [hunk 4]
```diff
@@ -69,83 +72,119 @@ local function IsValidString( str )
...
 end
 
--- Used to update a player's table of associated variables
-function cfcuser:UpdateUser( user, lastonline, factionid, kills, deaths, factionrank, hoursinfaction )
-    if not cfcuser:UserExists( user ) then return end
+
+--Used to update a player's table of associated variables
+function cfcuser:UpdateUser( user, lastonline, factionid, kills, deaths, factionrank )
+	local PlayerEnt = user
+	if ( not IsValid( PlayerEnt ) ) and ( not PlayerEnt:IsPlayer() ) then
```
Comment (sarahsturgeon) [inline lua/cfcfactions/core/sv/sv_users.lua]:
```suggestion
	if not ( IsValid( PlayerEnt ) and PlayerEnt:IsPlayer() ) then
```

File: lua/cfcfactions/core/sv/sv_users.lua [hunk 5]
```diff
@@ -69,83 +72,119 @@ local function IsValidString( str )
...
+    	LastOnline = cfcFactions:TimeStamp()
+    else
+    	userTable["LastOnline"] = lastonline
     end
-    if IsValidNumber( kills ) then userFactionTable["Kills"] = kills end
-    if IsValidNumber( deaths ) then userFactionTable["Deaths"] = deaths end
-    if IsValidString( factionrank ) then userFactionTable["FactionRank"] = factionrank end
-    if IsValidNumber( hoursinfaction ) and hoursinfaction > 0 then userFactionTable["HoursInFaction"] = hoursinfaction end
+
+    if ( not IsValidNumber( factionid ) ) then
```
Comment (sarahsturgeon) [inline lua/cfcfactions/core/sv/sv_users.lua]:
```suggestion
    if not IsValidNumber( factionid ) then 
```

File: lua/cfcfactions/core/sv/sv_users.lua [hunk 6]
```diff
@@ -69,83 +72,119 @@ local function IsValidString( str )
...
+		    	userFactionTable["Kills"] = kills
+		    end
+
+		    if IsValidNumber( deaths ) then 
+		    	userFactionTable["Deaths"] = deaths
+		    end
+
+		    if IsValidString( factionrank ) then
+		    	userFactionTable["FactionRank"] = factionrank
+		    end
```
Comment (sarahsturgeon) [inline lua/cfcfactions/core/sv/sv_users.lua]: These should be de-dented one level

File: lua/cfcfactions/core/sv/sv_users.lua [hunk 7]
```diff
@@ -69,83 +72,119 @@ local function IsValidString( str )
...
+		    if IsValidString( factionrank ) then
+		    	userFactionTable["FactionRank"] = factionrank
+		    end	
+    	else
+
+    	end
+
+    end  
+
+end
```
Comment (sarahsturgeon) [inline lua/cfcfactions/core/sv/sv_users.lua]: Could you add some comments in the empty conditionals saying what's supposed to (or what could) happen there?

**Review by sarahsturgeon — CHANGES_REQUESTED** [review]

**Comment by sarahsturgeon** [comment]
Spacing fixes proposed here: https://github.com/brandonsturgeon/cfc_factions_3/pull/61

File: lua/cfcfactions/core/sv/sv_users.lua (hunk 6 above)
Comment (ghost) [reply lua/cfcfactions/core/sv/sv_users.lua]: This looks proper on my end. Github makes it look like it has an extra indentation going on.

Comment (sarahsturgeon) [reply lua/cfcfactions/core/sv/sv_users.lua]: it's just because it's tabs - my whitespace pr should fix that up

**Pull request merged**